tests/components/nibe_heatpump/test_button.py

```diff
@@ -0,0 +1,91 @@
+"""Test the Nibe Heat Pump config flow."""
+from typing import Any
+from unittest.mock import AsyncMock, patch
+
+from nibe.coil import Coil
+from nibe.coil_groups import UNIT_COILGROUPS
+from nibe.heatpump import Model
+import pytest
+
+from homeassistant.components.button import DOMAIN as PLATFORM_DOMAIN, SERVICE_PRESS
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    STATE_UNAVAILABLE,
+    STATE_UNKNOWN,
+    Platform,
+)
+from homeassistant.core import HomeAssistant
+
+from . import async_add_and_get_entry
+
+MOCK_ENTRY_DATA = {
+    "model": None,
+    "ip_address": "127.0.0.1",
+    "listening_port": 9999,
+    "remote_read_port": 10000,
+    "remote_write_port": 10001,
+    "word_swap": True,
+    "connection_type": "nibegw",
+}
+
+
+@pytest.fixture(autouse=True)
+async def fixture_single_platform():
+    """Only allow this platform to load."""
+    with patch("homeassistant.components.nibe_heatpump.PLATFORMS", [Platform.BUTTON]):
+        yield
+
+
+@pytest.mark.parametrize(
+    ("model", "entity_id"),
+    [
+        (Model.F1155, "button.f1155_alarm_reset"),
+        (Model.S320, "button.s320_reset_alarm"),
+    ],
+)
+async def test_reset_button(
+    hass: HomeAssistant,
+    mock_connection: AsyncMock,
+    model: Model,
+    entity_id: str,
+    coils: dict[int, Any],
+):
+    """Test reset button."""
+
+    unit = UNIT_COILGROUPS[model.series]["main"]
+
+    # Setup a non alarm state
+    coils[unit.alarm_reset] = 0
+    coils[unit.alarm] = 0
+
+    coordinator = await async_add_and_get_entry(
+        hass, {**MOCK_ENTRY_DATA, "model": model.name}
+    )
+
+    state = hass.states.get(entity_id)
+    assert state
+    assert state.state == STATE_UNAVAILABLE
+
+    # Signal alarm
+    coils[unit.alarm] = 100
+    await coordinator.async_refresh()
```

Comment:
Move time forward instead as needed to trigger a coordinator refresh by time.

https://github.com/home-assistant/core/blob/534bb740699f25796ba4b8a248598fe885fb7a87/tests/common.py#L403-L406